addons/base/views.py

```diff
@@ -532,6 +534,44 @@ def addon_delete_file_node(self, target, user, event_type, payload):
                 file_node.delete(user=user)
 
 
+def get_storage_usage_of_children(folder):
+    folder_children = folder.children.all()
+    total_storage = 0
+    for item in folder_children:
+        if item.kind == 'file':
+            total_storage += item.versions.all().aggregate(sum=Sum('size'))['sum']
+        elif item.kind == 'folder':
+            total_storage += get_storage_usage_of_children(item)
+    return total_storage
+
+
+@file_signals.file_updated.connect
+def storage_usage_cache_control(self, target, user, event_type, payload):
+    if storage_usage_flag_active():
+        provider = payload.get('provider')
+
+        if provider == 'osfstorage':
+            path = payload['metadata']['path']
+            osfstorage = target.get_addon('osfstorage')
+            storage_usage = cache.get('storage_usage:' + osfstorage._id)
+
+            if not storage_usage:
+                storage_usage = osfstorage.storage_usage
+            elif event_type in ('file_added', 'file_updated'):
+                file_node = OsfStorageFile.objects.get(_id=path)
+                storage_usage += file_node.versions.first().size
+
+            if event_type == 'file_removed':
+                if path.endswith('/'):
+                    folder = TrashedFolder.objects.get(_id=path.rstrip('/'))
+                    storage_usage -= get_storage_usage_of_children(folder)
```

Comment:
I see what you mean about having to recursively sum the sizes of files within a folder.

In order to simplify this, how about you just invalidate the cache key when a folder is deleted?